opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/instances/TaskValueDelegate.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.provider.tasks.processors.instances;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import org.dmfs.jems.optional.decorators.Mapped;
+import org.dmfs.optional.NullSafe;
+import org.dmfs.provider.tasks.TaskDatabaseHelper;
+import org.dmfs.provider.tasks.model.ContentValuesInstanceAdapter;
+import org.dmfs.provider.tasks.model.CursorContentValuesTaskAdapter;
+import org.dmfs.provider.tasks.model.InstanceAdapter;
+import org.dmfs.provider.tasks.model.TaskAdapter;
+import org.dmfs.provider.tasks.processors.EntityProcessor;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.tasks.contract.TaskContract;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+/**
+ * An instance {@link EntityProcessor} which delegates to the appropriate task {@link EntityProcessor}.
+ *
+ * @author Marten Gajda
+ */
+public final class TaskValueDelegate implements EntityProcessor<InstanceAdapter>
+{
+
+    private final EntityProcessor<TaskAdapter> mDelegate;
+
+
+    public TaskValueDelegate(EntityProcessor<TaskAdapter> delegate)
+    {
+        mDelegate = delegate;
+    }
+
+
+    @Override
+    public InstanceAdapter insert(SQLiteDatabase db, InstanceAdapter entityAdapter, boolean isSyncAdapter)
+    {
+        // for now inserts are just forwarded, but we have to make sure we return an InstanceAdapter which contains the current instance row _ID.
+        // TODO: in case of a recurring master we should add an RDATE to it
+        // TODO: handle when inserting an override that already exists, either update the existing one or fail
+        TaskAdapter taskResult = mDelegate.insert(db, entityAdapter.taskAdapter(), false);
+        Cursor c = db.query(TaskDatabaseHelper.Tables.INSTANCES, new String[] { TaskContract.Instances._ID },
+                TaskContract.Instances.TASK_ID + "=" + taskResult.id(), null, null, null, null);
+        try
```

Comment:
`try-with-resource` opportunity here (and another one down below).

Reply:
done